lib/plug/conn.ex

```diff
@@ -460,12 +460,15 @@ defmodule Plug.Conn do
   otherwise `{:error, reason}`.
   """
   @spec chunk(t, body) :: {:ok, t} | {:error, term} | no_return
-  def chunk(%Conn{state: :chunked} = conn, ""), do: {:ok, conn}
   def chunk(%Conn{adapter: {adapter, payload}, state: :chunked} = conn, chunk) do
-    case adapter.chunk(payload, chunk) do
-      :ok                  -> {:ok, conn}
-      {:ok, body, payload} -> {:ok, %{conn | resp_body: body, adapter: {adapter, payload}}}
-      {:error, _} = error  -> error
+    case IO.iodata_length(chunk) do
```

Comment:
I did some pretty rudimentary benchmarks but I think :erlang.iolist_length is probably optimal.

It looks like iolist_size (the function behind iodata_length) is written in C https://github.com/erlang/otp/blob/b4ac8b2b32f094217d0533ee139273923c3a8af7/erts/preloaded/src/erlang.erl#L1070 it converts the iolist into a binary and then grabs its bytesize. 

And my benchmarks of various ways were not faster 

```
## BasicBench
benchmark name       iterations   average time 
IO.iodata_length       10000000   0.36 µs/op
IO.iodata_to_binary    10000000   0.49 µs/op
empty_iodata?          10000000   0.52 µs/op
```